modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/ast/MethodElementAST.java

```diff
@@ -131,25 +131,29 @@ public String getName() {
    * {@inheritDoc}
    */
   @Override
-  public Class getDeclaringClass() {
+  public Optional<Class<?>> getDeclaringClass() {
+    //TODO This will be removed soon.
     try {
-      return Class
-          .forName(processingEnvironment.getElementUtils().getBinaryName((TypeElement) method.getEnclosingElement()).toString());
+      return Optional.ofNullable(Class
+          .forName(processingEnvironment.getElementUtils().getBinaryName((TypeElement) method.getEnclosingElement()).toString()));
     } catch (ClassNotFoundException e) {
-      throw new RuntimeException(e);
+      //nothing
     }
+    return empty();
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public boolean equals(Object o) {
-    if (this == o)
+    if (this == o) {
       return true;
+    }
 
-    if (o == null || getClass() != o.getClass())
+    if (o == null || getClass() != o.getClass()) {
       return false;
+    }
```

Comment:
since we are here, the equalsbuilder is overkill, just do this.method.equals(that.method)

Reply:
fixed